### PR TITLE
[soil] Update repository

### LINKED ISSUE #16785: Package soil fetches from a dead website.
Error: Building package soil:x86-windows failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: soil:x86-windows
  Vcpkg version: 2021-01-13-d67989bce1043b98092ac45996a8230a059a2d7e

### PR DESCRIPTION
**Pull request by NancyLi1013**
**Describe the pull request**

- What does your PR fix? Fixes #16785 #15800

The website http://www.lonesock.net has been gone for some time. So we check the repository to https://github.com/littlstar/soil.

Note: No feature needs to test.

### CODE REVIEW

**Review by JackBoosY — APPROVED** [review]

**Comment by strega-nil** [comment]
Is there a reason that we aren't using https://github.com/paralin/soil ? it seems to be closer to the original, with a CMakeLists.txt (although it isn't very good...)

**Comment by NancyLi1013** [comment]
> Is there a reason that we aren't using https://github.com/paralin/soil ? it seems to be closer to the original, with a CMakeLists.txt (although it isn't very good...)

Since there is no any update for a long time(7 years ago) in this repository.

**Comment by strega-nil** [comment]
@NancyLi1013 there are no updates ever, since the library has been stable since the 2000s. Those are simply updates to either file names, or to build scripts.

**Comment by NancyLi1013** [comment]
Thanks for your informaton @strega-nil.

I will update to use the repository https://github.com/paralin/soil.

File: ports/soil/portfile.cmake [hunk 1]
```diff
@@ -1,28 +1,28 @@
...
+    REF 8bb18a909f94e58afbc0bda941ffc6eee58b4066 # committed on 2014-03-06
+    SHA512 6cbaa10d8b2a274c389fda723db53a3f1ba7d25a7367df40efec4d0553c09f0d67fb16f927bba2ff0aed4234e3a83922edcc574ffac72dd7e05d6cec768b561b
+    HEAD_REF master
+    PATCHES fix-cmakelists.patch
 )
 
 file(COPY
-    ${CMAKE_CURRENT_LIST_DIR}/CMakeLists.txt
     ${CMAKE_CURRENT_LIST_DIR}/SOILConfig.cmake.in
     ${CMAKE_CURRENT_LIST_DIR}/SOILConfigVersion.cmake.in
```
Comment (NancyLi1013) [inline ports/soil/portfile.cmake:14]:
```suggestion
    ${CMAKE_CURRENT_LIST_DIR}/soilConfig.cmake.in
    ${CMAKE_CURRENT_LIST_DIR}/soilConfigVersion.cmake.in
```

**Comment by BillyONeal** [comment]
Thanks for your help :D

**Pull request merged**